src/components/annual-report/report-data/section4/useFetchSectionFourData.js

```diff
@@ -0,0 +1,111 @@
+import { useDataEngine } from '@dhis2/app-runtime'
+import { useState } from 'react'
+
+const section4ByLevel = {
+    data_detail_by_level: {
+        resource: 'analytics.json',
+        params: ({ dataElements, orgUnits, orgUnitLevel, periods }) => ({
+            dimension: `dx:${dataElements.join(';')},ou:${
+                orgUnits.join(';') + ';' + orgUnitLevel
+            },pe:${periods.join(';')}`,
+        }),
+    },
+}
+
+const section4Overall = {
+    data_over_all_org_units: {
+        resource: 'analytics.json',
+        params: ({ dataElements, orgUnits, periods }) => ({
+            dimension: `dx:${dataElements.join(';')},ou:${orgUnits.join(
+                ';'
+            )},pe:${periods.join(';')}`,
+        }),
+    },
+}
+
+export const fetchDataByLevel = async ({ engine, variables }) => {
+    const dataBySubPeriod = await engine.query(section4ByLevel, {
+        variables,
+    })
+    return dataBySubPeriod
+}
+
+export const useFetchSectionFourData = () => {
+    const [loading, setLoading] = useState(false)
+    const [data, setData] = useState(null)
+    const [error, setError] = useState(null)
+
+    const engine = useDataEngine()
+
+    const refetch = async ({ variables = {} }) => {
+        const { denominatorRelations } = variables?.mappedConfiguration
+        const denominatorRelationDEs = [
+            ...denominatorRelations?.map((rel) => rel.A.id),
+            ...denominatorRelations?.map((rel) => rel.B.id),
+        ]
```

Comment:
My linter flagged these lines with "Unsafe usage of optional chaining. If it short-circuits with 'undefined' the evaluation will throw TypeError" -- here's a suggestion:
```suggestion
        const denominatorRelations = variables?.mappedConfiguration?.denominatorRelations
        const denominatorRelationDEs = denominatorRelations ? [
            ...denominatorRelations.map((rel) => rel.A.id),
            ...denominatorRelations.map((rel) => rel.B.id),
        ] : []
```

Reply:
Ah. Yeah. I see that optional chaining is kind of useless when rereading it 😄.

I guess it would make sense to address it even earlier in the code where these denominatorRelations get summarized into mappedConfiguration (https://github.com/hisprwanda/who-data-quality-annual-report/blob/933933c10bfeb54ffc9d25213d56c908194fb47a/src/utils/utils.js#L75-L76). Will also update that